x-pack/plugins/security_solution/server/usage/queries/utils/get_search_for_all_rules.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SearchRequest } from '@elastic/elasticsearch/lib/api/types';
+import type { AggregationsAggregationContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { RULE_EXECUTION_LOG_PROVIDER } from '../../../lib/detection_engine/rule_execution_log/event_log/constants';
+
+/**
+ * Given an aggregation of "aggs" this will return a search for all rules within 24 hours.
+ * @param eventLogIndex The event log index such as ".kibana-event-log-8.2.0*"
+ * @param aggs The aggregation to break things down by
+ */
+export const getSearchForAllRules = ({
+  eventLogIndex,
+  aggs,
+}: {
+  eventLogIndex: string;
+  aggs: Record<string, AggregationsAggregationContainer>;
+}): SearchRequest => ({
+  index: eventLogIndex,
+  size: 0,
+  track_total_hits: false,
+  aggs,
+  query: {
+    bool: {
+      filter: {
+        bool: {
+          must: [
+            {
+              range: {
+                '@timestamp': {
+                  gte: 'now-24h',
```

Comment:
No action is needed here, but I'm starting to wonder if we need to start finding the previous usage collector run timestamp and use that. I have noticed unusual collection cadences within some of the security clusters.

Reply:
Had the same question above 😅  https://github.com/elastic/kibana/pull/128216/files#r837923596

If consistency can't be guaranteed (runs same time each day, introducing drift), would be nice to have a service that pushes the right `day window` datetimes down for these queries to consume.